internal/core/data/v2/controller/http/event.go

```diff
@@ -26,15 +26,13 @@ import (
 )
 
 type EventController struct {
-	reader io.EventReader
-	dic    *di.Container
+	dic *di.Container
 }
 
 // NewEventController creates and initializes an EventController
 func NewEventController(dic *di.Container) *EventController {
 	return &EventController{
-		reader: io.NewEventRequestReader(),
```

Comment:
Why removed. Makes sense to only create once and use it. Looks like previous code didn't use it.

Reply:
Yeah, I thought previous code doesn't use this reader so simply removed it; however, you're right that we only need to create the reader once and then reuses it.  Changed the code to use reader from EventController.

Reply:
I thought it was redundant as previous code didn't use it; however, I agree with you that we should initialize the reader once and then reuse it.  Added the reader back to event controller.